dump/match_helper.py

```diff
@@ -0,0 +1,80 @@
+        sai_hostif_obj_key = ret["keys"][-1]
+        if sai_hostif_obj_key in ret["return_values"] and "SAI_HOSTIF_ATTR_OBJ_ID" in ret["return_values"][sai_hostif_obj_key]:
+            asic_port_obj_id = ret["return_values"][sai_hostif_obj_key]["SAI_HOSTIF_ATTR_OBJ_ID"]
+    return req, asic_port_obj_id, ret
+
+# Vlan Helper Methods
+
+def fetch_vlan_oid(match_engine, vlan_name, ns):
+    # Convert 'Vlanxxx' to 'xxx'
+    if vlan_name[0:4] != "Vlan" or not vlan_name[4:].isnumeric():
+        vlan_num = -1
+    else:
+        vlan_num = int(vlan_name[4:])
+
+    # Find the table named "ASIC_STATE:SAI_OBJECT_TYPE_VLAN:*" in which SAI_VLAN_ATTR_VLAN_ID = vlan_num
+    req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_VLAN", key_pattern="*", field="SAI_VLAN_ATTR_VLAN_ID", 
+                    value=str(vlan_num), ns=ns)
+    ret = match_engine.fetch(req)
+    vlan_oid = ""
+    if ret["keys"]:
+        vlan_oid = ret["keys"][0].split(":", 2)[-1]
+    return req, vlan_oid, ret
+
+# LAG Helper Methods
+
+def get_lag_members_from_cfg(match_engine, lag_name, ns):
+    """
+    Get the members associated with a LAG from Config DB
+    """
+    lag_members = []
+    req = MatchRequest(db="CONFIG_DB", table="PORTCHANNEL_MEMBER", key_pattern=lag_name + "|*", ns=ns)
+    ret = match_engine.fetch(req)
+    for key in ret["keys"]:
+        lag_members.append(key.split("|")[-1])
+    return req, lag_members, ret
+
+def get_lag_and_member_obj(match_engine, port_asic_obj, ns):
+    """
+    Given the member port oid, fetch lag_member & lag oid's
+    """
+    req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_LAG_MEMBER", key_pattern="*", field="SAI_LAG_MEMBER_ATTR_PORT_ID",
+                        value=port_asic_obj, return_fields=["SAI_LAG_MEMBER_ATTR_LAG_ID"], ns=ns)
+    ret = match_engine.fetch(req)
+    lag_member_key = ""
+    lag_oid = ""
+    if not ret["error"] and ret["keys"]:
+        lag_member_key = ret["keys"][-1]
+        if lag_member_key in ret["return_values"] and "SAI_LAG_MEMBER_ATTR_LAG_ID" in ret["return_values"][lag_member_key]:
+            lag_oid = ret["return_values"][lag_member_key]["SAI_LAG_MEMBER_ATTR_LAG_ID"]
+    return lag_member_key, lag_oid
+
+def fetch_lag_oid(match_engine, lag_name, ns):
+    """
+    Finding the relevant SAI_OBJECT_TYPE_LAG key directly from the ASIC is not possible given a LAG name
+    Thus, using the members to find SAI_LAG_MEMBER_ATTR_LAG_ID
+    """
+    _, lag_members, _ = get_lag_members_from_cfg(match_engine, lag_name, ns)
+    lag_type_oids = set()
+    for port_name in lag_members:
+        _, port_asic_obj, _ = fetch_port_oid(match_engine, port_name, ns)
+        if port_asic_obj:
+            lag_member_key, lag_oid = get_lag_and_member_obj(match_engine, port_asic_obj, ns)
+            lag_type_oids.add(lag_oid)
+    lag_type_oid, lag_type_oids = "", list(lag_type_oids)
+    if lag_type_oids:
+        lag_type_oid = lag_type_oids[-1]
```

Comment:
Can you elaborate in comment why oid is the last pair in the list?

Reply:
Handled #

Reply:
If there is no ambiguity, then either of them will be same. If no, then I’m choosing a random option. So first or last should be fine, I guess